repo-scripts/size-analysis/size-analysis.test.ts

```diff
@@ -0,0 +1,401 @@
+    ).to.equal(0);
+    expect(
+      memberList.enums.filter(each => each.localeCompare(replaceTo) === 0)
+        .length
+    ).to.equal(0);
+  });
+});
+
+describe('test mapSymbolToType helper function', () => {
+  it('test if function correctly categorizes symbols that are misplaced', () => {
+    let memberList: MemberList = {
+      functions: ['aVar', 'bFunc', 'cFunc'],
+      classes: ['bClass', 'cClass'],
+      variables: ['aClass', 'bVar', 'cVar', 'aEnum'],
+      enums: ['bEnum', 'cEnum', 'dEnum', 'aFunc'],
+      externals: []
+    };
+
+    const map: Map<string, string> = new Map([
+      ['aFunc', 'functions'],
+      ['bFunc', 'functions'],
+      ['aClass', 'classes'],
+      ['bClass', 'classes'],
+      ['aVar', 'variables'],
+      ['bVar', 'variables'],
+      ['aEnum', 'enums']
+    ]);
+
+    memberList = mapSymbolToType(map, memberList);
+
+    expect(memberList.functions).to.have.members(['aFunc', 'bFunc', 'cFunc']);
+    expect(memberList.functions).to.not.include.members(['aVar']);
+    expect(memberList.classes).to.have.members(['aClass', 'bClass', 'cClass']);
+    expect(memberList.variables).to.not.include.members(['aClass', 'aEnum']);
+    expect(memberList.variables).to.have.members(['aVar', 'bVar', 'cVar']);
+    expect(memberList.enums).to.have.members([
+      'aEnum',
+      'bEnum',
+      'cEnum',
+      'dEnum'
+    ]);
+    expect(memberList.enums).to.not.include.members(['aFunc']);
+
+    expect(memberList.functions).to.have.length(3);
+    expect(memberList.classes).to.have.length(3);
+    expect(memberList.variables).to.have.length(3);
+    expect(memberList.enums).to.have.length(4);
+  });
+});
+
+describe('test writeReportToFile helper function', () => {
+  it('should throw error when given path exists and points to directory', () => {
+    const aDir = resolve('./a-dir/a-sub-dir');
+    fs.mkdirSync(aDir, { recursive: true });
+    expect(() => writeReportToFile('content', aDir)).to.throw(
+      ErrorCode.OUTPUT_FILE_REQUIRED
+    );
+  });
+
+  it('should not throw error when given path does not pre-exist', () => {
+    const aPathToFile = resolve('./a-dir/a-sub-dir/a-file');
+    expect(() => writeReportToFile('content', aPathToFile)).to.not.throw();
+    fs.unlinkSync(aPathToFile);
+  });
+  after(() => {
+    fs.rmdirSync('a-dir', { recursive: true });
+  });
+});
+
+describe('test writeReportToDirectory helper function', () => {
+  it('should throw error when given path exists and points to a file', () => {
+    const aDir = resolve('./a-dir/a-sub-dir');
+    fs.mkdirSync(aDir, { recursive: true });
+    const aFile = `a-file`;
+    const aPathToFile = `${aDir}/${aFile}`;
+    fs.writeFileSync(aPathToFile, 'content');
+    expect(() =>
+      writeReportToDirectory('content', aFile, aPathToFile)
+    ).to.throw(ErrorCode.OUTPUT_DIRECTORY_REQUIRED);
+  });
+
+  it('should not throw error when given path does not pre-exist', () => {
+    const aDir = resolve('./a-dir/a-sub-dir');
```

Comment:
This relies on the assumption that the tests run in order where the previous test creates the dir, which may not be true. This test should create the folder in its own body. The previous test should clean up once it's finished.